dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -290,9 +298,27 @@ namespace dwa_local_planner {
     cmd_vel.twist.linear.y = drive_cmds.pose.position.y;
     cmd_vel.twist.angular.z = tf2::getYaw(drive_cmds.pose.orientation);
 
-    //if we cannot move... tell someone
     std::vector<geometry_msgs::PoseStamped> local_plan;
     if(path.cost_ < 0) {
+      // no valid path found; slow down respecting acceleration limits and report NO_VALID_CMD
+      double vx = robot_vel.pose.position.x;
+      double vy = robot_vel.pose.position.y;
+      double vt = tf2::getYaw(robot_vel.pose.orientation);
+      const base_local_planner::LocalPlannerLimits& limits = planner_util_.getCurrentLimits();
+      cmd_vel.twist.linear.x = vx - std::copysign(limits.acc_lim_x / controller_frequency_, vx);
+      cmd_vel.twist.linear.y = vy - std::copysign(limits.acc_lim_y / controller_frequency_, vy);
+      cmd_vel.twist.angular.z = vt - std::copysign(limits.acc_lim_theta / controller_frequency_, vt);
+
+      // clip reduced velocities to zero, so we don't start moving in the opposite direction
+      if (sign(cmd_vel.twist.linear.x) != sign(vx))
+        cmd_vel.twist.linear.x = 0.0;
+      if (sign(cmd_vel.twist.linear.y) != sign(vy))
+        cmd_vel.twist.linear.y = 0.0;
+      if (sign(cmd_vel.twist.angular.z) != sign(vt))
+        cmd_vel.twist.angular.z = 0.0;
+      ROS_DEBUG_NAMED("dwa_local_planner", "Slowing down from (%.2f, %.2f, %.2f) to (%.2f, %.2f, %.2f)",
+                      vx, vy, vt, cmd_vel.twist.linear.x, cmd_vel.twist.linear.y, cmd_vel.twist.angular.z);
```

Comment:
Q: I have some reservations about this: 
- I thought sending a [0,0,0] command just results in the base controller shorting the motors, so the more aggressive motor configuration in the firmware shouldn't change anything regarding sending a [0,0,0] velocity command? CC @bhuvanchandra 
- For this sort of acceleration control, I think the DWA planner runs a bit at a low frequency
- Do we do this to avoid tipping over? If so, might also be better to use feedback control (i.e. detect toppling with IMU and reduce deceleration if so; but again, this should probably run at a higher frequency)
- Maybe we need a different parameter for this deceleration (which would also allow to disable the feature) / deceleration in general instead of using `limits.acc_lim_x`?

Reply:
Also, if you make an obstacle suddenly appear in front of the robot, isn't it the velocity limiter that sends the [0,0,0] command? (if not, shouldn't we enable the velocity limiter?)

Reply:
> Also, if you make an obstacle suddenly appear in front of the robot, isn't it the velocity limiter that sends the [0,0,0] command? (if not, shouldn't we enable the velocity limiter?)

yes,,, but that's fine, as we are in an emergency, so no point to be smooth

Reply:
> yes,,, but that's fine, as we are in an emergency, so no point to be smooth

What I am trying to say is: if your test scenario was `Here I run the robot on a straight route and kick fast in front.`, shouldn't the velocity limiter have stopped the robot (in which case, the changes in this PR wouldn't apply)?